gdynet/preprocess.py

```diff
@@ -169,6 +169,33 @@ def construct_graph(self, traj_coords, lattices, atom_types, target_index):
                     'target_index': target_index,
                     'nbr_lists': nbr_lists,
                     'nbr_dists': nbr_dists}
+        elif self.backend == 'ndirect':
+            stcs = [Structure(lattice=lattices[i],
+                              species=atom_types,
+                              coords=traj_coords[i],
+                              coords_are_cartesian=True)
+                    for i in tqdm(range(len(traj_coords)),
+                                  desc='Generating structure...', disable=not self.verbose)]
+            a, b, c = [np.ceil(2*self.radius/d).astype('int')
+                       for d in stcs[0].lattice.abc]
+            if [a, b, c] != [1, 1, 1]:
+                _ = [stc.make_supercell(
+                    [np.ceil(2*self.radius/d).astype('int') for d in stc.lattice.abc])
+                    for stc in tqdm(stcs, desc='Building supercell...', disable=not self.verbose)]
+            nbr_lists = np.array([stc.distance_matrix.argsort()[
+                                 :, 1:1+self.n_nbrs] for stc in tqdm(
+                stcs, desc='Generating neighbor index...', disable=not self.verbose)], dtype='int32')
+            nbr_dists = np.array([np.sort(stc.distance_matrix)[
+                                 :, 1:1+self.n_nbrs] for stc in tqdm(
```

Comment:
Can you reformat your code according to PEP8? There should be whitespaces between 1+ for example.